sympy/physics/mechanics/joint.py

```diff
@@ -0,0 +1,681 @@
+
+        self.child_joint_point.set_pos(self.parent_joint_point,
+                                       x_x * self.parent.frame.x +
+                                       x_y * self.parent.frame.y)
+        self.child_joint_point.set_vel(self.parent.frame,
+                                       v_x * self.parent.frame.x +
+                                       v_y * self.parent.frame.y)
+
+        self.child.masscenter.v2pt_theory(self.parent.masscenter,
+                                          self.parent.frame, self.child.frame)
+
+
+class SphericalJoint(Joint):
+    """
+    Spherical (Ball) Joint.
+
+    It is defined such that the child body rotates in any direction with respect
+    to the parent body about the point of spherical joint through the angle
+    theta_x, theta_y, theta_z along x, y and z directions of the parent's frame.
+    The point of joint (pin joint) is defined by two points in each body which
+    coincides together. They are supplied as parent_point_pos and child_point_pos.
+    In addition, the child's reference frame can be arbitrarily rotated a constant
+    amount with respect to the parent axis by passing an axis in child body which
+    must align with parent axis after the rotation.
+
+    Parameters
+    ----------
+    name: String
+        Name of the joint which makes it unique. Should be different from other
+        joints.
+    parent: Body
+        Instance of Body class serving as the parent in the joint.
+    child: Body
+        Instance of Body class serving as the child in the joint.
+    parent_point_pos: 3 Tuple, Optional
+        Defines the joint's point where the parent will be connected to child.
+        3 Tuple defines the values of x, y and z directions w.r.t parent's
+        frame. If it is not supplied, center of mass is added as default.
+    child_point_pos: 3 Tuple, Optional
+        Defines the joint's point where the child will be connected to parent.
+        3 Tuple defines the values of x, y and z directions w.r.t child's frame.
+        If it is not supplied, center of mass is added as default.
+
+    Examples
+    --------
+    Adds spherical Joint between parent's masscenter and a point at unit distance
+    along child's y axis.
+
+    >>> from sympy import Symbol
+    >>> from sympy.physics.mechanics import Body, SphericalJoint
+    >>> parent = Body('parent')
+    >>> child = Body('child')
+    >>> l = Symbol('l')
+    >>> spherical_joint = SphericalJoint('sphericaljoint', parent, child, \
+                                         child_point_pos=(0, l, 0))
+    >>> spherical_joint.coordinates
+    [sphericaljoint_alpha(t), sphericaljoint_beta(t), sphericaljoint_gamma(t)]
+    >>> spherical_joint.speeds
+    [sphericaljoint_omega_alpha(t), sphericaljoint_omega_beta(t), sphericaljoint_omega_gamma(t)]
+
+    """
+    def __init__(self, name, parent, child, parent_point_pos=None,
+                 child_point_pos=None, rot_order=None):
```

Comment:
Isn't doing that again the guidelines? I remember we changed some_parameter='something' to some_parameter=None in pydy.